### ISSUE TITLE
JSON schema value description overrides property description in tooltip

### ISSUE THREAD

**Issue by kryops** [microsoft/vscode#202145]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.85.1
- OS Version: macOS 14.2.1

Steps to Reproduce:

1. Create a JSON schema where both the property of an object and the value type it references have a `description`:

```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$ref": "#/definitions/Root",
  "definitions": {
    "Root": {
      "type": "object",
      "properties": {
        "foo": {
          "$ref": "#/definitions/Expression",
          "description": "Property description of foo"
        }
      },
      "additionalProperties": false
    },
    "Expression": {
      "type": "string",
      "pattern": "^=.+$",
      "description": "Description of the Expression type"
    }
  }
}
```

2. Create a JSON file matching the schema:

```json
{
  "foo": "=bar"
}
```

3. Hover over the property and the value, checking the tooltip

Expected behavior:
- Hovering over the **value** displays the **value description** as tooltip (or both the value and property description)
- Hovering over the **property** displays the **property description** as tooltip (or both the property and value description)

Actual behavior:
- Hovering over the **value** displays the **value description** as tooltip
- Hovering over the **property** displays the **value description** as tooltip as well

![image](https://github.com/microsoft/vscode/assets/1042594/d3bb8a37-3c2b-4e1f-8821-55dab5c08505)

Thanks for looking into it!

**Comment by aeschli** [comment]
I don't think that you can say that the first object describes the property, while the second one the value.
A $ref just mean that the content of the object is defined somewhere else. Typically you would only have a `$ref` in that object and nothing else.
When resolving a $ref, the referenced object is merged into the object that contains the $ref, overwriting existing properties.

No changes planed.
